src/frontend/src/planner/recursive_union.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::binder::{BoundSetExpr, ShareId};
+use crate::error::Result;
+use crate::optimizer::plan_node::LogicalRecursiveUnion;
+use crate::{PlanRef, Planner};
+
+impl Planner {
+    pub(super) fn plan_recursive_union(
+        &mut self,
+        base: BoundSetExpr,
+        recursive: BoundSetExpr,
+        id: ShareId,
+    ) -> Result<PlanRef> {
```

Comment:
What about change the signature to `fn plan_recursive_union(&mut self, recursiveUnion: RecursiveUnion)`?

Reply:
the `id` here needs to be passed in `plan_share`, so it at least needs to include an extra `id` field in addition to the `RecursiveUnion` - maybe keeping the current signature is clear enough?